### PR TITLE
DP-7579 : Utility Nav Component.

### PR DESCRIPTION
**Pull request by smurrayatwork**
**Description:** 
This adds the UtilityNav organism to the storybook, as well as support for adding its markdown info for display. The atoms SvgBuilding and SvgLogin were also added in order to properly render the navigation icons. In the future, this component will need to be expanded to handle new icons.

**Jira:** 
https://jira.state.ma.us/browse/DP-7579

**To Test:**
- [ ] Pull the branch and run 'npm run storybook'.
- [ ] Browse to http://localhost:6006/ (or the port that you chose instead) and verify there's now an UtilityNav link on the left side under Organisms. Click on it and verify that the component displayed matches the attached screenshot.
- [ ] Verify that each knob changes the corresponding component.

**Post Deploy Steps:**
N/A

**Relevant Screenshots/GIFs:**
screen shot 2018-02-16 at 2 59 21 pm

**Additional Notes:**
N/A
---

[Peer Review Checklist](docs/Peer%20review%20checklist.md)

### CODE REVIEW

File: src/components/organisms/UtilityNav/UtilityNav.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+### Pattern Contains
+* Utility Panel
+
+### JavaScript Used
+* Utility Nav (js/modules/utilNav.js)
+
+### Variables
+~~~
+utilityNav: {
```
Comment (iansholtys) [inline src/components/organisms/UtilityNav/UtilityNav.md]: No need the `utilityNav` variable here.

File: src/components/organisms/UtilityNav/UtilityNav.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+### Description
+Sitewide utility links
+
+### Status
+* Stable as of 5.0.0
```
Comment (iansholtys) [inline src/components/organisms/UtilityNav/UtilityNav.md]: Not sure if the Status or JavaScript Used sections are necessary. Don't think we ever discussed these though.

File: src/components/organisms/UtilityPanel/index.js [hunk 3]
```diff
@@ -6,17 +6,17 @@ import DecorativeLink from '../../atoms/links/DecorativeLink';
 
 const UtilityPanel = (utilityPanel) => {
   const descriptionClasses = ['ma__utility-panel__description'];
-  if (!utilityPanel.links) {
+  const links = (typeof utilityPanel.links === 'undefined' || utilityPanel.links === null) ? [] : utilityPanel.links;
```
Comment (iansholtys) [inline src/components/organisms/UtilityPanel/index.js]: Isn't the case of `links` being undefined or null taken care of by the defaultProp of `links: []`?

Comment (smurrayatwork) [reply src/components/organisms/UtilityPanel/index.js]: It should be, but when this was not done this way, it was null.

File: src/components/organisms/UtilityNav/UtilityNav.stories.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      };
+      Object.keys(iconComponents).forEach((key) => {
+        icons[key] = key;
+      });
+      const storyProps = {
+        text: text(`utilityNav.text.${itemIndex}`, item.text),
+        ariaLabelText: text(`utilityNav.ariaLabelText.${itemIndex}`, item.ariaLabelText),
+        closeText: text(`utilityNav.closeText.${itemIndex}`, item.closeText),
+        panel: object(`utilityNav.panel.${itemIndex}`, item.panel)
+      };
```
Comment (avrilpearl) [inline src/components/organisms/UtilityNav/UtilityNav.stories.js]: @smurrayatwork i think you are missing items in some of these name as that is the prop these are all passed to - right?

Comment (avrilpearl) [reply src/components/organisms/UtilityNav/UtilityNav.stories.js]: i.e. utilityNav.items.text.${itemIndex}

File: src/components/organisms/UtilityNav/UtilityNav.stories.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      const comp = select(`utilityNav.icons.${itemIndex}`, icons, item.icon.type.name);
+      storyProps.icon = iconComponents[comp];
+      return(storyProps);
+    });
+    return(<UtilityNav {...newProps} />);
```
Comment (avrilpearl) [inline src/components/organisms/UtilityNav/UtilityNav.stories.js]: @smurrayatwork - is there a reason why the google language props are not available in knobs ui for this component?

File: src/components/organisms/UtilityNav/index.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  }
+  onClick(divId, e) {
+    e.preventDefault();
+    this.setState({
+      navSelected: (this.state.navSelected === -1) ? divId : -1
+    });
+  }
+  render() {
+    const { navSelected } = this.state;
+    return((
```
Comment (avrilpearl) [inline src/components/organisms/UtilityNav/index.js]: i think you should have `const utilityNav = this.props` declared here --> update respective call to this.props below

File: src/components/organisms/UtilityNav/index.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  /** An array of navigation items to display to the user. */
+  items: PropTypes.arrayOf(PropTypes.shape({
+    /** The text to display for the main navigation item. */
+    text: PropTypes.string.isRequired,
+    /** Defines the label to use with aria-label. */
+    ariaLabelText: PropTypes.string,
+    /** The icon to display to the left of text. */
+    icon: PropTypes.oneOfType([
+      PropTypes.shape(SvgBuilding.propTypes),
+      PropTypes.shape(SvgLogin.propTypes)
```
Comment (avrilpearl) [inline src/components/organisms/UtilityNav/index.js]: cool! - good to know this is a way to approach this

File: src/components/organisms/UtilityNav/UtilityNav.md (hunk 2 above)
Comment (avrilpearl) [reply src/components/organisms/UtilityNav/UtilityNav.md]: we have not, will need to go back and resolve this post deadlines - will put it on the board

**Pull request merged**